src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -884,7 +878,7 @@ protected virtual IDbDataParameter CreateDatabaseParameter(IDbCommand command, D
         /// </summary>
         /// <param name="logEvent">Current logevent.</param>
         /// <param name="parameterInfo">Parameter configuration info.</param>
-        protected internal virtual object GetParameterValue(LogEventInfo logEvent, DatabaseParameterInfo parameterInfo)
+        protected internal virtual object CreateDatabaseParameterValue(LogEventInfo logEvent, DatabaseParameterInfo parameterInfo)
```

Comment:
IMO this is more a retrieve/get then a create. It won't create (`new`) things?

Reply:
It will create a value for all situations except when extracted rawvalue matches dbtype

Reply:
It's a getOrCreate ;)

Not sure if we should have a virtual getOrCreate (need to check that)

IMO a create method should never "get”, so "get" is s better name (I'm trying to keep it easy to implement this method by others as it's virtual)

Reply:
Renamed to GetDatabaseParameterValue